libcloud/test/compute/test_retry_limit.py

```diff
@@ -0,0 +1,75 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import socket
+import tempfile
+from libcloud.utils.py3 import httplib
+
+from mock import Mock, patch, MagicMock
+
+from libcloud.common.base import Connection
+from libcloud.common.base import Response
+from libcloud.common.exceptions import RateLimit
+from libcloud.test import unittest
+
+CONFLICT_RESPONSE_STATUS = [
+    ('status', '429'), ('reason', 'CONFLICT'),
+    ('retry_after', '3'),
+    ('content-type', 'application/json')]
+SIMPLE_RESPONSE_STATUS = ('HTTP/1.1', 429, 'CONFLICT')
+
+
+class RateLimitTestCase(unittest.TestCase):
+
+    def _raise_socket_error(self):
+        raise socket.gaierror('')
+
+    def test_retry_connection(self):
```

Comment:
I was just merging the PR and cleaning some minor style things and noticed that this method doesn't actually test the retry mechanism.

You mock the `request` method which means you are testing the mock and not the actual request method which performs a retry.

Can you please work on some actual tests for the retry mechanism inside `request` method?